test/functional/settings_controller_test.rb

```diff
@@ -20,4 +22,10 @@ class SettingsControllerTest < ActionController::TestCase
     assert_equal("You must be logged in to access this page", flash[:error])
     assert_response :redirect
   end
+
+  test "test index method with out school code" do
+    sign_in @user3
+    get :index, params: { settings: { contact_email: "test@gmail.com", school: { id: @school3.id } } }
+    assert_response :success
```

Comment:
Would be great to start making slightly more detailed tests here — like checking the title of the page, looking in the HTML for the school name, for example.